entity/templates/src/main/java/package/web/rest/_EntityResource.java

```diff
@@ -40,15 +41,10 @@
 public class <%= entityClass %>Resource {
 
     private final Logger log = LoggerFactory.getLogger(<%= entityClass %>Resource.class);
-
-    @Inject
-    private <%= entityClass %>Repository <%= entityInstance %>Repository;<% if (dto == 'mapstruct') { %>
-
-    @Inject
-    private <%= entityClass %>Mapper <%= entityInstance %>Mapper;<% } %><% if (searchEngine == 'elasticsearch') { %>
-
-    @Inject
-    private <%= entityClass %>SearchRepository <%= entityInstance %>SearchRepository;<% } %>
+    <% var viaService = service != 'no'; -%>
+    <%- include('../../common/inject_template', {viaService: viaService}); -%><%
+    var instanceType = (dto == 'mapstruct') ? entityClass + 'DTO' : entityClass;
+    var instanceName = (dto == 'mapstruct') ? entityInstance + 'DTO' : entityInstance; -%>
 
```

Comment:
All done, too sleepy now ill update the pr tomorrow
On 26 Nov 2015 20:28, "Deepu K Sasidharan" <d4udts@gmail.com> wrote:

> I got the error let me take a look, its while running java integration
> test
> On 26 Nov 2015 15:46, "Julien Dubois" <notifications@github.com> wrote:
>
>> In entity/templates/src/main/java/package/web/rest/_EntityResource.java
>> <https://github.com/jhipster/generator-jhipster/pull/2310#discussion_r45950474>
>> :
>>
>> > @@ -40,15 +41,10 @@
>> >  public class <%= entityClass %>Resource {
>> >
>> >      private final Logger log = LoggerFactory.getLogger(<%= entityClass %>Resource.class);
>> > -
>> > -    @Inject
>> > -    private <%= entityClass %>Repository <%= entityInstance %>Repository;<% if (dto == 'mapstruct') { %>
>> > -
>> > -    @Inject
>> > -    private <%= entityClass %>Mapper <%= entityInstance %>Mapper;<% } %><% if (searchEngine == 'elasticsearch') { %>
>> > -
>> > -    @Inject
>> > -    private <%= entityClass %>SearchRepository <%= entityInstance %>SearchRepository;<% } %>
>> > +    <% var viaService = service != 'no'; -%>
>> > +    <%- include('../../common/inject_template', {viaService: viaService}); -%><%
>> > +    var instanceType = (dto == 'mapstruct') ? entityClass + 'DTO' : entityClass;
>> > +    var instanceName = (dto == 'mapstruct') ? entityInstance + 'DTO' : entityInstance; -%>
>> >
>>
>> No idea why they felt, I just ran them and got all them as errors -> I'll
>> check if you can't reproduce
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/jhipster/generator-jhipster/pull/2310/files#r45950474>
>> .
>>
>